src/main/java/net/minecraftforge/client/event/EntityViewRenderEvent.java

```diff
@@ -83,23 +90,67 @@ public void setDensity(float density)
     }
 
     /**
-     * Event that allows any feature to customize the rendering of fog.
+     * This event allows for customization of parameters related to fog rendering. The plane distances are based on the player's render distance.
+     *
+     * A FogMode of FOG_SKY is the skybox's fog.
+     * A FogMode of FOG_TERRAIN is what the fog induced by render distance uses. This works best for reducing the camera's visibility.
      */
-    @HasResult
-    public static class RenderFogEvent extends FogEvent
+    public static class RenderFogEvent extends EntityViewRenderEvent
```

Comment:
These are _technically_ breaking changes:
 - The [javadoc for `Event#setResult`](https://github.com/MinecraftForge/EventBus/blob/6d3407ccf8a8f74bc66c759cbf40f3909c918c8f/src/main/java/net/minecraftforge/eventbus/api/Event.java#L120-L127) states that attempting to set the result for an event which does not have one (i.e. no `@HasResult`) causes an `IllegalArgumentException`. Regardless of the actual visible implementation, that is the specification in the javadocs and we must abide by it. 

   Since there is already no use for the event result, you can keep the annotation and mark it with a `// TODO: ` to remove the annotation in 1.19.

 - The change in superclass now means that any possibly mod which expect `RenderFogEvent` to extend `FogEvent` will fail. While this doesn't sound like an issue, practically speaking, it is still technically a breaking change. For now, it should extend `FogEvent` and marked with a `// TODO: ` to change its superclass to `EntityViewRenderEvent` in 1.19.